MapboxNavigation/NavigationMapView.swift

```diff
@@ -383,16 +400,53 @@ open class NavigationMapView: MGLMapView {
                     layer.identifier != arrowLayerIdentifier && layer.identifier != arrowSymbolLayerIdentifier && layer.identifier != arrowCasingSymbolLayerIdentifier && layer.identifier != arrowLayerStrokeIdentifier && layer.identifier != waypointCircleIdentifier {
                     style.insertLayer(line, below: layer)
                     style.insertLayer(lineCasing, below: line)
-                    return
+                    break
                 }
             }
         }
+        guard routes.count > 1 else { return }
+        var tmpRoutes = routes
+        tmpRoutes.remove(at: activeRouteIndex)
+        guard let alternateRoute = tmpRoutes.first else { return }
+        
+        self.routes = routes
+        
+        let polyline = MGLPolylineFeature(coordinates: alternateRoute.coordinates!, count: alternateRoute.coordinateCount)
+        polyline.attributes[routeIndexAttribute] = routes.index(of: alternateRoute)
```

Comment:
Once this method runs to completion, the map will contain features that are associated with an index into a temporary array that has gone out of scope. There is no guarantee that this index means anything past this point.

NavigationMapView could hold onto its displayed routes in a property, perhaps in a dictionary that maps Routes to feature IDs or vice versa. Better yet, we can use Turf methods to find the route that comes closest to the tap point without relying on feature querying or indices.